tests/shared/src/test/scala/cats/effect/std/RandomSpec.scala

```diff
@@ -17,38 +17,238 @@
+          bytes2 <- random2.nextBytes(256)
+        } yield bytes1.length == 128 && bytes2.length == 256
+      }
+
+      "prevent array reference from leaking in ThreadLocalRandom.nextBytes impl" in real {
+        val random = Random.javaUtilConcurrentThreadLocalRandom[IO]
+        val nextBytes = random.nextBytes(128)
+        for {
+          bytes1 <- nextBytes
+          bytes2 <- nextBytes
+        } yield bytes1 ne bytes2
+      }
+
+      "prevent array reference from leaking in ScalaRandom.nextBytes impl" in real {
+        for {
+          random <- Random.scalaUtilRandom[IO]
+          nextBytes = random.nextBytes(128)
+          bytes1 <- nextBytes
+          bytes2 <- nextBytes
+        } yield bytes1 ne bytes2
+      }
     }
 
-    "prevent array reference from leaking in ThreadLocalRandom.nextBytes impl" in real {
-      val random = Random.javaUtilConcurrentThreadLocalRandom[IO]
-      val nextBytes = random.nextBytes(128)
-      for {
-        bytes1 <- nextBytes
-        bytes2 <- nextBytes
-      } yield bytes1 ne bytes2
+    "nextDouble" >> {
+      "generate random double values between 0.0 and 1.0" in real {
+        val numIterations: Int = 1000
+        for {
+          random <- Random.scalaUtilRandom[IO]
+          randomDoubles <- (1 to numIterations).toList.traverse(_ => random.nextDouble)
+        } yield randomDoubles.forall(double => double >= 0.0 && double < 1.0)
+      }
     }
 
-    "prevent array reference from leaking in ScalaRandom.nextBytes impl" in real {
-      for {
-        random <- Random.scalaUtilRandom[IO]
-        nextBytes = random.nextBytes(128)
-        bytes1 <- nextBytes
-        bytes2 <- nextBytes
-      } yield bytes1 ne bytes2
+    "nextFloat" >> {
+      "generate random float values between 0.0 and 1.0" in real {
+        val numIterations: Int = 1000
+        for {
+          random <- Random.scalaUtilRandom[IO]
+          randomFloats <- (1 to numIterations).toList.traverse(_ => random.nextFloat)
+        } yield randomFloats.forall(float => float >= 0.0f && float < 1.0f)
+      }
+    }
+
+    "nextGaussian" >> {
+      "generate random Gaussian distributed double values with mean 0.0 and standard deviation 1.0" in real {
+        val sampleSize = 1000
+        for {
+          random <- Random.scalaUtilRandom[IO]
+          gaussians <- (1 to sampleSize).toList.traverse(_ => random.nextGaussian)
+          mean = gaussians.sum / sampleSize
+          variance = gaussians.map(x => math.pow(x - mean, 2)).sum / sampleSize
+          stddev = math.sqrt(variance)
+        } yield math.abs(mean) < 0.1 && math.abs(stddev - 1.0) < 0.1
+      }
+    }
+
+    "nextInt" >> {
+      "generate random int value" in real {
+        for {
+          random <- Random.scalaUtilRandom[IO]
+          int <- random.nextInt
+        } yield int match {
+          case _: Int => true
+          case _ => false
+        }
```

Comment:
Is this really asserting something that is not guaranteed by the type system? :)

A no-op assert is fine for this. Just executing `nextInt` asserts that it links and runs on Scala.js and Native which is always useful.

```suggestion
          int <- random.nextInt
        } yield ok
```

Reply:
I see, it makes completely sense. I am going update it!